### PR TITLE
pflip etc

### PR DESCRIPTION
**Pull request by cscherrer**

### CODE REVIEW

File: gumbel-max-sampling.pdf
Comment (BaxterEaves) [inline gumbel-max-sampling.pdf:1]: did you intend to commit these docs?

File: src/misc/func.rs [hunk 1]
```diff
@@ -173,8 +173,38 @@ fn catflip(cws: &[f64], r: f64) -> Option<usize> {
     }
 }
 
+pub fn pflip(weights: &[f64], rng: &mut impl Rng) -> usize {
```
Comment (BaxterEaves) [inline src/misc/func.rs:176]: Can we implement this with a `sum: Option<f64>` argument? We often know the sum of the weights, so we often only need one scan to compute the sum of a predicate.

```rust
pub fn pflip(weights: &[f64], sum: Option<f64>, rng: &mut impl Rng) -> usize {
    let u = sum.unwrap_or_else(|| weights.iter().sum::<f64>()) * rng.gen::<f64>();
    weights.iter().map(|w| if w < u { 1 } else { 0 }).sum::<usize>() // is this count right?
}
```

**Review by BaxterEaves — CHANGES_REQUESTED** [review]
Looks great. Have a couple minor suggestions. Thanks so much for addressing the typos!

File: src/misc/func.rs [hunk 2]
```diff
@@ -173,8 +173,38 @@ fn catflip(cws: &[f64], r: f64) -> Option<usize> {
...
+
+//     let r = rng.sample(u) * scale;
+//     match catflip(&cws, r) {
+//         Some(ix) => ix,
+//         None => {
+//             let wsvec = weights.to_vec();
+//             panic!("Could not draw from {:?}", wsvec)
+//         }
+//     }
+// }
```
Comment (BaxterEaves) [inline src/misc/func.rs:204]: Can we remove the commented code?

File: gumbel-max-sampling.pdf
Comment (cscherrer) [reply gumbel-max-sampling.pdf:1]: Yes, but mostly just so we have them _somewhere_. Maybe better in Notion?

**Pull request merged**